WebExample/tests/input.spec.ts

```diff
@@ -0,0 +1,46 @@
+import {test, expect} from '@playwright/test';
+import type {Page} from '@playwright/test';
+import * as CONSTANTS from '../../constants';
+
+const setupInput = async (page: Page, mode: 'clear' | 'reset') => {
+  const inputLocator = await page.locator(`div#MarkdownInput_Example`);
+  await page.click(`[data-testid="${mode}"]`);
+
+  return inputLocator;
+};
+
+test.beforeEach(async ({page}) => {
+  await page.goto('http://localhost:19006/', {waitUntil: 'load'});
+});
+
+test.describe('standard input behaviour', () => {
+  test('standard input results', async ({page}) => {
+    const inputLocator = await setupInput(page, 'clear');
+
+    await inputLocator.pressSequentially(CONSTANTS.EXAMPLE_CONTENT);
+    const value = await inputLocator.innerText();
+    expect(value).toEqual(CONSTANTS.EXAMPLE_CONTENT);
+  });
+
+  test('fast type cursor position', async ({page}) => {
+    const EXAMPLE_LONG_CONTENT = CONSTANTS.EXAMPLE_CONTENT.repeat(3);
+
+    const inputLocator = await setupInput(page, 'clear');
+
+    await inputLocator.pressSequentially(EXAMPLE_LONG_CONTENT);
+
+    expect(await inputLocator.innerText()).toBe(EXAMPLE_LONG_CONTENT);
+
+    const cursorPosition = await page.evaluate(() => {
+      const editableDiv = document.querySelector('div[contenteditable="true"]') as HTMLElement;
+      const range = window.getSelection()?.getRangeAt(0);
+      if (!range || !editableDiv) return null;
+      const preCaretRange = range.cloneRange();
+      preCaretRange.selectNodeContents(editableDiv);
+      preCaretRange.setEnd(range.endContainer, range.endOffset);
+      return preCaretRange.toString().length;
```

Comment:
Checking cursor position seems to be a pretty common case in live markdown testing. In the future, I think we will add more cases that check if caret is correctly positioned. Let's move it to the function in utils so it can be used everywhere

Reply:
Basically it's a bit annoying to pass any variables to page.evaluate, especially objects and functions, and since I wasn't sure how far those function will be developed I just decided to clean this up later.